pubtools/_quay/operator_pusher.py

```diff
@@ -522,6 +505,112 @@ def build_index_images(self):
                     )
                 )
                 failed_items[id(item)] = True
+        return items_opted_in, failed_items
+
+    def _get_non_fbc_items_for_version(self, items, version, items_opted_in):
+        """Return non fbc items for given ocp version.
+
+        Args:
+            items: List[ContainerPushItem]
+                list of push items
+            version: str
+                ocp version for which items should be returned.
+            items_opted_in: Dict[int, bool]
+                list of push items opted in fbc.
+        Returns List[ContainerPushItem]:
+            list of items not opted in fbc
+        """
+        non_fbc_items = []
+        osev_tuple = tuple([int(x) for x in version.replace("v", "").split(".")])
+        for item in items:
+            if not items_opted_in[id(item)] or (items_opted_in[id(item)] and osev_tuple <= (4, 12)):
+                non_fbc_items.append(item)
+            elif items_opted_in[id(item)] and osev_tuple >= (4, 13):
+                LOG.warning(
+                    "Skipping {i}".format(i=item)
+                    + "from iib build as it's opted in for FBC and targeting OCP version >=4.13"
+                )
+        return non_fbc_items
+
+    def _create_item_groups_for_version(
+        self, non_fbc_items, version, is_hotfix=False, is_prerelease=False
+    ):
+        """Iterate thought non fbc items and group those together based on version.
+
+        Args:
+            non_fbc_items: List[ContainerPushItem]
+                list of items not opted in fbc
+            version: str
+                ocp version
+            is_hotfix: bool
+                flag indicating items are for hotfix push
+            is_prerelease: bool
+                flag indicating items are for prerelease push
+        Returns Dict[str, Dict[str, Any]]:
+            Dictionary of items grouped by ocp version
+        """
+        item_groups = {
+            version: {
+                "items": [],
+                "overwrite": True,
+                "build_tags": [],
```

Comment:
This is always empty? Later in the code when it's accessed, it seems that it's always an empty list

Reply:
it's used to store "{0}-{1}".format(index_image.split(":")[1], self.task_id)

Reply:
That's later in the code, but why is it even a part of this dictionary?

Reply:
now I understand what you mean. Now it's not needed as I removed build tag for hotfix